staging/cse/windows/azurecnifunc.ps1

```diff
@@ -494,6 +494,79 @@ function GenerateAzureStackCNIConfig
     Set-ItemProperty -Path $azureCNIConfigFile -Name IsReadOnly -Value $true
 }
 
+
+function GetIpv4AddressFromParsedContent
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        $ParsedContent
+    )
+
+    if ($ParsedContent[0].ipv4 -and $ParsedContent[0].ipv4.ipAddress -and $ParsedContent[0].ipv4.ipAddress.Count -gt 0)
+    {
+        return $ParsedContent[0].ipv4.ipAddress[0].privateIpAddress
+    }
+    else
+    {
+        return $null
+    }
+}
+
+function GetIpv6AddressFromParsedContent
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        $ParsedContent
+    )
+
+    if ($ParsedContent[0].ipv6 -and $ParsedContent[0].ipv6.ipAddress -and $ParsedContent[0].ipv6.ipAddress.Count -gt 0)
+    {
+        return $ParsedContent[0].ipv6.ipAddress[0].privateIpAddress
+    }
+    else
+    {
+        return $null
+    }
+}
+
+function GetMetadataContent
+{
+    $Retries = 10
+    $RetryDelaySeconds = 15

```

Comment:
I think 15 seconds is too much. It may significantly increase provisioning time. It's on the same node, so I would spam requests multiple times a second

Reply:
Will change this - have gone with 120 retries at 1 per second. That's 2 minutes which should be fine.